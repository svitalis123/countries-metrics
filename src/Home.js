import React from 'react';

function Home() {
  return (
    <div>
      hello home
    </div>
  );
}

export default Home;
