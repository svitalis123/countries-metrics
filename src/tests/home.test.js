import { render } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Home from '../Home';
import store from '../redux/configureStore';

describe('test home', () => {
  it('test home component', () => {
    const testmore = render(
      <React.StrictMode>
        <Provider store={store}>
          <BrowserRouter>
            <Home />
          </BrowserRouter>
        </Provider>
      </React.StrictMode>,
    );
    expect(testmore).toMatchSnapshot();
  });
});
