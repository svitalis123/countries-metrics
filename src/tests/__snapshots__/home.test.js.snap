// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test home test home component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="home_container"
      >
        <div
          class="home_container_world_div"
        >
          <figure
            class="home_container_world_div_figure"
          >
            <img
              alt="world"
              src="/assets/world_map.svg"
            />
          </figure>
          <h2>
            The Whole World
          </h2>
        </div>
        <div>
          <select
            class="home_container_items_select"
          >
            <option
              value="All"
            >
              Select Continent
            </option>
            <option
              value="Africa"
            >
              Africa
            </option>
            <option
              value="Asia"
            >
              Asia
            </option>
            <option
              value="Antarctic"
            >
              Antarctic
            </option>
            <option
              value="Europe"
            >
              Europe
            </option>
            <option
              value="North America"
            >
              North America
            </option>
            <option
              value="Oceania"
            >
              Oceania
            </option>
            <option
              value="South America"
            >
              South America
            </option>
          </select>
          <div
            class="home_container_items_div"
          >
            <div
              class="homemap_container"
            >
              <figure>
                <a
                  class="homemap_container_divs_inside_p"
                  href="/more"
                >
                  →
                </a>
                <img
                  alt="Africa"
                  src="/assets/Africa.svg"
                />
                <h3>
                  Africa
                </h3>
              </figure>
            </div>
            <div
              class="homemap_container"
            >
              <figure>
                <a
                  class="homemap_container_divs_inside_p"
                  href="/more"
                >
                  →
                </a>
                <img
                  alt="Asia"
                  src="/assets/Asia.svg"
                />
                <h3>
                  Asia
                </h3>
              </figure>
            </div>
            <div
              class="homemap_container"
            >
              <figure>
                <a
                  class="homemap_container_divs_inside_p"
                  href="/more"
                >
                  →
                </a>
                <img
                  alt="Europe"
                  src="/assets/Europe.svg"
                />
                <h3>
                  Europe
                </h3>
              </figure>
            </div>
            <div
              class="homemap_container"
            >
              <figure>
                <a
                  class="homemap_container_divs_inside_p"
                  href="/more"
                >
                  →
                </a>
                <img
                  alt="Oceania"
                  src="/assets/Oceania.svg"
                />
                <h3>
                  Oceania
                </h3>
              </figure>
            </div>
            <div
              class="homemap_container"
            >
              <figure>
                <a
                  class="homemap_container_divs_inside_p"
                  href="/more"
                >
                  →
                </a>
                <img
                  alt="North America"
                  src="/assets/North America.svg"
                />
                <h3>
                  North America
                </h3>
              </figure>
            </div>
            <div
              class="homemap_container"
            >
              <figure>
                <a
                  class="homemap_container_divs_inside_p"
                  href="/more"
                >
                  →
                </a>
                <img
                  alt="South America"
                  src="/assets/South America.svg"
                />
                <h3>
                  South America
                </h3>
              </figure>
            </div>
            <div
              class="homemap_container"
            >
              <figure>
                <a
                  class="homemap_container_divs_inside_p"
                  href="/more"
                >
                  →
                </a>
                <img
                  alt="Antarctic"
                  src="/assets/Antarctica.svg"
                />
                <h3>
                  Antarctic
                </h3>
              </figure>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="home_container"
    >
      <div
        class="home_container_world_div"
      >
        <figure
          class="home_container_world_div_figure"
        >
          <img
            alt="world"
            src="/assets/world_map.svg"
          />
        </figure>
        <h2>
          The Whole World
        </h2>
      </div>
      <div>
        <select
          class="home_container_items_select"
        >
          <option
            value="All"
          >
            Select Continent
          </option>
          <option
            value="Africa"
          >
            Africa
          </option>
          <option
            value="Asia"
          >
            Asia
          </option>
          <option
            value="Antarctic"
          >
            Antarctic
          </option>
          <option
            value="Europe"
          >
            Europe
          </option>
          <option
            value="North America"
          >
            North America
          </option>
          <option
            value="Oceania"
          >
            Oceania
          </option>
          <option
            value="South America"
          >
            South America
          </option>
        </select>
        <div
          class="home_container_items_div"
        >
          <div
            class="homemap_container"
          >
            <figure>
              <a
                class="homemap_container_divs_inside_p"
                href="/more"
              >
                →
              </a>
              <img
                alt="Africa"
                src="/assets/Africa.svg"
              />
              <h3>
                Africa
              </h3>
            </figure>
          </div>
          <div
            class="homemap_container"
          >
            <figure>
              <a
                class="homemap_container_divs_inside_p"
                href="/more"
              >
                →
              </a>
              <img
                alt="Asia"
                src="/assets/Asia.svg"
              />
              <h3>
                Asia
              </h3>
            </figure>
          </div>
          <div
            class="homemap_container"
          >
            <figure>
              <a
                class="homemap_container_divs_inside_p"
                href="/more"
              >
                →
              </a>
              <img
                alt="Europe"
                src="/assets/Europe.svg"
              />
              <h3>
                Europe
              </h3>
            </figure>
          </div>
          <div
            class="homemap_container"
          >
            <figure>
              <a
                class="homemap_container_divs_inside_p"
                href="/more"
              >
                →
              </a>
              <img
                alt="Oceania"
                src="/assets/Oceania.svg"
              />
              <h3>
                Oceania
              </h3>
            </figure>
          </div>
          <div
            class="homemap_container"
          >
            <figure>
              <a
                class="homemap_container_divs_inside_p"
                href="/more"
              >
                →
              </a>
              <img
                alt="North America"
                src="/assets/North America.svg"
              />
              <h3>
                North America
              </h3>
            </figure>
          </div>
          <div
            class="homemap_container"
          >
            <figure>
              <a
                class="homemap_container_divs_inside_p"
                href="/more"
              >
                →
              </a>
              <img
                alt="South America"
                src="/assets/South America.svg"
              />
              <h3>
                South America
              </h3>
            </figure>
          </div>
          <div
            class="homemap_container"
          >
            <figure>
              <a
                class="homemap_container_divs_inside_p"
                href="/more"
              >
                →
              </a>
              <img
                alt="Antarctic"
                src="/assets/Antarctica.svg"
              />
              <h3>
                Antarctic
              </h3>
            </figure>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
